src/nvim/api/buffer.c

```diff
@@ -170,20 +192,34 @@ void buffer_set_line_slice(Buffer buffer,
                       ArrayOf(String) replacement,
                       Error *err)
 {
```

Comment:
doxygen would be better because it is the docs that an external client/plugin writer would read.